### PR TITLE
Feature/static necroplant

### PR DESCRIPTION
**Pull request by RickyVimon**
What's new?

- Necroplants now have a checkbox Static which, if checked and "Alive" checkbox is also checked, the necroplant can be placed anywhere in the level and will not burry and teleport to other positions.
- Static necroplants will respawn again if the players also respawns (either on a checkpoint or at the beggining).
- 3 static Necroplants added during the Level 2 scene.

### CODE REVIEW

File: Scripts/src/Script/EnemyManager.cpp [hunk 1]
```diff
@@ -359,4 +363,12 @@ bool EnemyManager::IsEnemyMeleeAttacking(EnemyController* enemy)
 bool EnemyManager::IsEnemyRangedAttacking(EnemyController* enemy)
 {
 	return static_cast<Mushdoom*>(enemy)->ranged_attack;
+}
+
+void EnemyManager::ResetStaticNecroplants()
+{
+	for (unsigned i = 0; i < static_necroplants.size(); ++i)
```
Comment (raulgonzalezupc) [inline Scripts/src/Script/EnemyManager.cpp:370]: unsigned int

**Review by raulgonzalezupc — COMMENTED** [review]
LGTM, but you have to change some things

File: Scripts/src/Script/EnemyManager.h [hunk 2]
```diff
@@ -44,17 +45,21 @@ class EnemyManager : public Script
...
 	bool IsEnemyMeleeAttacking(EnemyController* enemy);
 	bool IsEnemyRangedAttacking(EnemyController* enemy);
 
 
+
 public:
 	std::vector<EnemyController*> enemies;
 	unsigned current_number_of_enemies_alive = 0;
 
+	std::vector<Necroplant*> static_necroplants;
```
Comment (raulgonzalezupc) [inline Scripts/src/Script/EnemyManager.h:61]: I think you need to call this delete on the destructor

Comment (RickyVimon) [reply Scripts/src/Script/EnemyManager.h:61]: Changes applied

File: Scripts/src/Script/EnemyManager.cpp (hunk 1 above)
Comment (RickyVimon) [reply Scripts/src/Script/EnemyManager.cpp]: done, thanks :)

**Review by raulgonzalezupc — APPROVED** [review]

**Review by Pacasasgar — APPROVED** [review]
LGTM, be careful with the scene conflicts when merging, take only one and then change that one.

**Pull request merged**